### ISSUE TITLE
 E30 M3 Tach output solution

### ISSUE THREAD

**Issue by rusefi** [rusefi/rusefi#907]
At the moment we have a total mess of dizzy output and tach output, and potentially another unique request for E30 M3:

```Orchardperformance 11:48 AM
General question for anyone that has used the Tach output...
Just trying to solve the niggles with my E30 installation so I can run it more as a daily and I need to get the Tech working, Its a 12v high low signal so it's going on U604.
The signal needs to be a 50% duty cycle pulsed 3 times a rev like the dizzy output would give. (2 pulses for a 4 pot and 3 for a 6 pot.)
Is there a way to set the dizzy output to a duty cycle? Or is there a way to set the tach output to duty cycle and multiplied x3?
I suspect I can set this up via FSIO but with all the pre-canbus BMW stuff using this setup then it might be worth baking this into the standard ECU options.
Orchardperformance 12:16 PM
Actually having the Tach output as a multiplier and duty cycle setup may make a lot of sense for many conversions. A friend just pointed out it's a nightmare for the E30 M3 as there is only a 4 cylinder dashboard unique to the car so they often need the ECU running 6 or 8 coils but with a 4 cylinder output.
I will try to take a look at the feasibility later on```

I believe this should all be simplified by leveraging FSIO.
At the moment FSIO allows duty cycle control but not programmatic frequency control. It could be that FSIO programmatic frequency control is the way to address this requirement.

**Comment by shadowm60** [comment]
I have tried an implementation as follows:
Added a new input filed for the tacho that will need the number of pulses per crank rotation.
with this info and current RPM we can calculate the pulse with that needs to output.

What i did not took into consideration is the field pulse duration, if we do have more then one pulse it will actually generate a PWM 50% duty.

Should i keep the pulse duration as well? what if the configured pulse duration is not possible since the number is to high?

**Comment by rusefi** [comment]
@shadowm60 I am sorry I am not sure what is the question exactly. 

I believe there could be at least two typos - while I am probably able to guess, this complicates comprehension. If you are referencing any existing configuration fields please spell out the exact names of the existing name(s) of the existing configuration fields.

To be honest I do not exactly remember what we have in the code base right now. First step could be to document the behaviors which are possible at the moment and to analyze if we shall start by simplifying/unifying existing behavior. Maybe the first PR would be just adding human-readable overview of the current status of this area into tachometer.cpp?

Maybe chatting in Slack could help maybe not. I am not sure what is the best approach here :(

**Comment by shadowm60** [comment]
Sorry for that, it is late here, i have created a pull request, it is simpler to follow

**Comment by OrchardPerformance** [comment]
@shadowm60 - Can you confirm if this is now working?

**Comment by shadowm60** [comment]
Yes, it can be closed, changes implemented in #1203
